app-backup/dar/dar-2.7.2.ebuild

```diff
@@ -0,0 +1,90 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit flag-o-matic
+
+DESCRIPTION="A full featured backup tool, aimed for disks"
+HOMEPAGE="http://dar.linux.free.fr/"
+SRC_URI="mirror://sourceforge/dar/${P}.tar.gz"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~ppc ~sparc ~x86 ~amd64-linux"
+IUSE="argon2 curl dar32 dar64 doc gcrypt gpg lzo nls rsync threads xattr"
+
+REQUIRED_USE="?? ( dar32 dar64 )
+		gpg? ( gcrypt )"
+
+RESTRICT="test" # need to be run as root
+
+RDEPEND="
+	app-arch/bzip2:=
+	app-arch/xz-utils:=
+	sys-libs/libcap
+	>=sys-libs/zlib-1.2.3:=
+	argon2? ( app-crypt/argon2 )
+	curl? ( net-misc/curl )
+	gcrypt? ( dev-libs/libgcrypt:0= )
+	gpg? ( app-crypt/gpgme )
+	lzo? ( dev-libs/lzo:= )
+	rsync? ( net-libs/librsync:= )
+	threads? ( dev-libs/libthreadar:= )
+	xattr? ( sys-apps/attr:= )
+"
```

Comment:
My app-portage/iwdevtools are nagging
```
 * QA: detected possibly incorrect RDEPEND (app-backup/dar-2.7.2)
 * app-arch/xz-utils:=    | app-arch/xz-utils
 *                        > app-arch/zstd:=
 * app-crypt/argon2       | app-crypt/argon2:=
 * app-crypt/gpgme        | app-crypt/gpgme:1=
 *                        > dev-libs/libgpg-error
 * dev-libs/libthreadar:= | dev-libs/libthreadar
 * dev-libs/lzo:2=        | dev-libs/lzo:2
 * sys-apps/attr:=        | sys-apps/attr
```
Basically saying:
1. some of these := aren't necessary (no subslot), harmless but customary not to define if unneeded
2. directly linked with libzstd but no dependency, this looks like automagic so needs a USE for `--disable-libzstd-linking`
3. directly linked with libgpg-error but not explicitly listed, this is only with USE=gcrypt (not that it's a big issue considering gcrypt depends on this, but that's a rabbit hole better avoided).
4. argon2 and gpgme define a subslot but here the := is missing, the 1 in :1= isn't *essential* in this case but if want to be pedantic (given it defines a slot).

Reply:
Nice tool, thanks.  I've used it and fixed the libs.  For gpg I also had to add `dev-libs/libassuan`, also a dependency of `app-crypt/gpgme`.

I also added **zstd** enabled by default, without a USE flag (it's been a pull request about this [here](https://github.com/gentoo/gentoo/pull/21558) closed and dropped by now, but it seemed like a good idea).  But I added a flag for **lz4** instead - it was automagically detected too.

Reply:
Good thinking wrt zstd USE. I agree with it but wasn't really thinking about it when I suggested.

libgpg-error wasn't for gpg (well not only for it), as I mentioned gcrypt uses it:
```
[ebuild  N     ] app-backup/dar-2.7.2::gentoo  USE="gcrypt -argon2 -curl -dar32 -dar64 -doc -gpg -lz4 -lzo -nls -rsync -threads -xattr"
<snip>
 * QA: detected possibly incorrect RDEPEND (app-backup/dar-2.7.2)
 *  > dev-libs/libgpg-error
```
gpg may need it as well but given there's a REQUIRED_USE it only needs to be listed once behind `gcrypt?`

wrt libassuan, it doesn't actually use it and it's just a bad inject by gpgme discarded by -Wl,--as-needed, in these case I'd say it's okay to not list it. If say.. gpgme ever stopped using it and libassuan is depcleaned, it won't break dar. While for libgpg-error you'd get an unresolved library and wouldn't startup anymore without preserve-libs.
```
[ebuild   R    ] app-backup/dar-2.7.2::gentoo  USE="gcrypt gpg -argon2 -curl -dar32 -dar64 -doc -lz4 -lzo -nls -rsync -threads -xattr"
<snip>
 * QA: detected possibly incorrect RDEPEND (app-backup/dar-2.7.2)
 * dev-libs/libassuan < 
```
Or was it actually in NEEDED for you? I guess it's possible there's some other condition.

Reply:
I don't have --as-needed.  It looks like this:
`LDFLAGS="-Wl,--defsym=__gentoo_check_ldflags__=0"`

But if I added it libassuan went away.

I'll remove libassuan and move `dev-libs/libgpg-error` on `gcrypt?`

Reply:
Ah I see, it'd be good to use LDFLAGS="${LDFLAGS} -Wl,--defsym=__gentoo_check_ldflags__=0" to keep the defaults.